src/Microsoft.Health.Fhir.Core/Features/Operations/IIntegrationDataStoreClientInitilizer.cs

```diff
@@ -3,27 +3,55 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
-using System.Threading;
+using System;
 using System.Threading.Tasks;
 using Microsoft.Health.Fhir.Core.Configs;
 
 namespace Microsoft.Health.Fhir.Core.Features.Operations
 {
-    public interface IIntegrationDataStoreClientInitilizer<T>
+    public interface IIntegrationDataStoreClientInitilizer<TBlobServiceClient, TBlockBlobClient, TBlobClient>
     {
         /// <summary>
         /// Used to get a client that is authorized to talk to the integration data store.
         /// </summary>
-        /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>A client of type T</returns>
-        Task<T> GetAuthorizedClientAsync(CancellationToken cancellationToken);
+        Task<TBlobServiceClient> GetAuthorizedClientAsync();
 
         /// <summary>
         /// Used to get a client that is authorized to talk to the integration data store.
         /// </summary>
         /// <param name="integrationDataStoreConfiguration">Integration dataStore configuration</param>
-        /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>A client of type T</returns>
-        Task<T> GetAuthorizedClientAsync(IntegrationDataStoreConfiguration integrationDataStoreConfiguration, CancellationToken cancellationToken);
+        Task<TBlobServiceClient> GetAuthorizedClientAsync(IntegrationDataStoreConfiguration integrationDataStoreConfiguration);
+
+        /// <summary>
+        /// Used to get a block blob client that is authorized to talk to the integration data store.
+        /// </summary>
+        /// <param name="blobUri">A blob uri.</param>
+        /// <returns>A block blob client of type TBlobClient</returns>
```

Comment:
We create/get a blob reference in 3 ways.
1) CloudBlobClient -> get a container ref -> get a blob ref. (doable in v12)
2) CloudBlobClient.GetBlobReferenceFromServerAsync. (no equivalent in v12)
3) CloudBlob ctor passing a blob uri and CloudBlobClient instance. (no equivalent in v12)

For 2 and 3 in v12, a credential or a connection string/container name/blob name must be passed to BlobClient's ctor.  That requires creating a credential object by parsing a connection string or parsing a blob uri to get a container and blob name. Parsing a connection string to create a credential object seems easier than parsing a blob uri.